src/aspire/abinitio/commonline_base.py

```diff
@@ -77,22 +78,34 @@ def _build(self):
             logger.error(msg)
             raise NotImplementedError(msg)
 
+    @property
+    def pf(self):
+        if self._pf is None:
+            self._pf = self._prepare_pf()
+        return self._pf
+
+    def _prepare_pf(self):
+        """
+        Prepare the polar Fourier transform used for correlations.
+        """
         imgs = self.src.images[:]
 
         if self.mask:
             fuzz_mask = fuzzy_mask((self.n_res, self.n_res), self.dtype)
             imgs = imgs * fuzz_mask
 
         # Obtain coefficients of polar Fourier transform for input 2D images
-        self.pft = PolarFT(
+        pft = PolarFT(
             (self.n_res, self.n_res), self.n_rad, self.n_theta, dtype=self.dtype
         )
-        self.pf = self.pft.transform(imgs)
+        pf = pft.transform(imgs)
 
         # We remove the DC the component. pf has size (n_img) x (n_theta/2) x (n_rad-1),
         # with pf[:, :, 0] containing low frequency content and pf[:, :, -1] containing
         # high frequency content.
-        self.pf = self.pf[:, :, 1:]
+        pf = pf[:, :, 1:]
+
+        return pf
```

Comment:
Since you are using the class for the `self.src.images` data, might as well just have this function assign to `self._pf`, and update L84.

Reply:
Ah yeah, good point. Done. Thanks!